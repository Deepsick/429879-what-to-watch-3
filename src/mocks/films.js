export const movies = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    picture: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    name: `Bohemian Rhapsody`,
    picture: `bohemian-rhapsody.jpg`,
  },
  {
    name: `Macbeth`,
    picture: `macbeth.jpg`,
  },
  {
    name: `Aviator`,
    picture: `aviator.jpg`,
  },
  {
    name: `We need to talk about Kevin`,
    picture: `we-need-to-talk-about-kevin.jpg`,
  },
  {
    name: `What We Do in the Shadows`,
    picture: `what-we-do-in-the-shadows.jpg`,
  },
  {
    name: `Revenant`,
    picture: `revenant.jpg`,
  },
  {
    name: `Johnny English`,
    picture: `johnny-english.jpg`,
  },
];

export const movie = {
  name: `The Grand Budapest Hotel`,
  genre: `drama`,
  year: 2014,
};
