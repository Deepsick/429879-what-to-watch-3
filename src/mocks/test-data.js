export const movie = {
  name: `The Grand Budapest Hotel`,
  genre: `drama`,
  year: 2014,
};

export const onMovieTitleClick = () => {};
export const onCardHover = () => {};
export const onMouseOut = () => {};
export const mockFunction = () => {};
export const hoverDelay = 1000;
export const isShown = true;
export const moviesCount = 8;
export const muted = true;
export const isPlaying = true;
export const genre = `drama`;
export const trailer = `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const isVideo = false;
export const mockBool = true;
export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export const movies = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    picture: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    id: `1`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    name: `Bohemian Rhapsody`,
    picture: `bohemian-rhapsody.jpg`,
    id: `2`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    name: `Macbeth`,
    picture: `macbeth.jpg`,
    id: `3`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    name: `Aviator`,
    picture: `aviator.jpg`,
    id: `4`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    name: `We need to talk about Kevin`,
    picture: `we-need-to-talk-about-kevin.jpg`,
    id: `5`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    name: `What We Do in the Shadows`,
    picture: `what-we-do-in-the-shadows.jpg`,
    id: `6`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    name: `Revenant`,
    picture: `revenant.jpg`,
    id: `7`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    name: `Johnny English`,
    picture: `johnny-english.jpg`,
    id: `8`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];

export const detailedMovies = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `drama`,
    id: `1`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody.jpg`,
    genre: `drama`,
    id: `2`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Macbeth`,
    poster: `macbeth.jpg`,
    genre: `drama`,
    id: `3`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Aviator`,
    poster: `aviator.jpg`,
    genre: `drama`,
    id: `4`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `We need to talk about Kevin`,
    poster: `we-need-to-talk-about-kevin.jpg`,
    genre: `drama`,
    id: `5`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `What We Do in the Shadows`,
    poster: `what-we-do-in-the-shadows.jpg`,
    genre: `drama`,
    id: `6`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Revenant`,
    poster: `revenant.jpg`,
    genre: `drama`,
    id: `7`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Johnny English`,
    poster: `johnny-english.jpg`,
    genre: `comedy`,
    id: `8`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
];


