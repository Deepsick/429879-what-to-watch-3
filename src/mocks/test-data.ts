export const mockFunction = () => {};
export const hoverDelay = 1000;
export const isShown = true;
export const moviesCount = 8;
export const muted = true;
export const isPlaying = true;
export const duration = 1000;
export const rating = 12345;
export const genre = `drama`;
export const trailer = `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const isVideo = false;
export const id = 1;
export const mockString = ``;
export const Auth = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};
export const user = {
  name: 'user',
  avatar: 'hello.jpg',
}
export const review = {
  rating: 8,
  comment: 'hello',
};
export const error = 'Невозможно получить данные';
export const comments = [
  {
    id: 1,
    comment: '123213',
    user: {
      id: 3,
      name: 'asdw',
    },
    rating: 111,
    date: '123',
  },
  {
    id: 2,
    comment: 'hi',
    user: {
      id: 5,
      name: 'asdwssss',
    },
    rating: 6664,
    date: '123',
  },
]

export const mockBool = true;
export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export const movies = [
  {
    bgColor: '#ffffff',
    preview: ``,
    video: ``,
    rating: 3,
    scoresCount: 300,
    duration: 100500,
    isFavorite: true,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `drama`,
    id: 1,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    bgColor: '#ffffff',
    preview: ``,
    video: ``,
    rating: 3,
    scoresCount: 300,
    duration: 100500,
    isFavorite: true,
    name: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody.jpg`,
    genre: `drama`,
    id: 2,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    bgColor: '#ffffff',
    preview: ``,
    video: ``,
    rating: 3,
    scoresCount: 300,
    duration: 100500,
    isFavorite: true,
    name: `Macbeth`,
    poster: `macbeth.jpg`,
    genre: `drama`,
    id: 3,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    bgColor: '#ffffff',
    preview: ``,
    video: ``,
    rating: 3,
    scoresCount: 300,
    duration: 100500,
    isFavorite: true,
    name: `Aviator`,
    poster: `aviator.jpg`,
    genre: `drama`,
    id: 4,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    bgColor: '#ffffff',
    preview: ``,
    video: ``,
    rating: 3,
    scoresCount: 300,
    duration: 100500,
    isFavorite: true,
    name: `We need to talk about Kevin`,
    poster: `we-need-to-talk-about-kevin.jpg`,
    genre: `drama`,
    id: 5,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    bgColor: '#ffffff',
    preview: ``,
    video: ``,
    rating: 3,
    scoresCount: 300,
    duration: 100500,
    isFavorite: true,
    name: `What We Do in the Shadows`,
    poster: `what-we-do-in-the-shadows.jpg`,
    genre: `drama`,
    id: 6,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    bgColor: '#ffffff',
    preview: ``,
    video: ``,
    rating: 3,
    scoresCount: 300,
    duration: 100500,
    isFavorite: true,
    name: `Revenant`,
    poster: `revenant.jpg`,
    genre: `drama`,
    id: 7,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    bgColor: '#ffffff',
    preview: ``,
    video: ``,
    rating: 3,
    scoresCount: 300,
    duration: 100500,
    isFavorite: true,
    name: `Johnny English`,
    poster: `johnny-english.jpg`,
    genre: `comedy`,
    id: 8,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
];


