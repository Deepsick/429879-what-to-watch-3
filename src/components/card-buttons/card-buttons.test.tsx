import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import CardButtons from './card-buttons';
import {id, mockFunction} from '../../mocks/test-data';

const isFavorite = true;
const isAddReview = true;

it(`Should render CardButtons component for film page correctly`, () => {
  const node = renderer.create(
      <BrowserRouter>
        <CardButtons
          id={id}
          isAddReview={isAddReview}
          isFavorite={isFavorite}
          onClick={mockFunction}
        />
      </BrowserRouter>
  ).toJSON();

  expect(node).toMatchSnapshot();
});

it(`Should render CardButtons component for main page correctly`, () => {
  const node = renderer.create(
      <BrowserRouter>
        <CardButtons
          id={id}
          isAddReview={!isAddReview}
          isFavorite={isFavorite}
          onClick={mockFunction}
        />
      </BrowserRouter>
  ).toJSON();

  expect(node).toMatchSnapshot();
});

it(`Should render CardButtons component for favorite film correctly`, () => {
  const node = renderer.create(
      <BrowserRouter>
        <CardButtons
          id={id}
          isAddReview={isAddReview}
          isFavorite={isFavorite}
          onClick={mockFunction}
        />
      </BrowserRouter>
  ).toJSON();

  expect(node).toMatchSnapshot();
});

it(`Should render CardButtons component for non-favorite film correctly`, () => {
  const node = renderer.create(
      <BrowserRouter>
        <CardButtons
          id={id}
          isAddReview={isAddReview}
          isFavorite={!isFavorite}
          onClick={mockFunction}
        />
      </BrowserRouter>
  ).toJSON();

  expect(node).toMatchSnapshot();
});

