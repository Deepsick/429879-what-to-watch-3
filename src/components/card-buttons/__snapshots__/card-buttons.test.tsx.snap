// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CardButtons component for favorite film correctly 1`] = `
<div
  className="movie-card__buttons"
>
  <a
    className="btn btn--play movie-card__button"
    href="/player/1"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </a>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="14"
      viewBox="0 0 18 14"
      width="18"
    >
      <use
        xlinkHref="#in-list"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
  <a
    className="btn movie-card__button"
    href="/films/1/review"
    onClick={[Function]}
  >
    Add review
  </a>
</div>
`;

exports[`Should render CardButtons component for film page correctly 1`] = `
<div
  className="movie-card__buttons"
>
  <a
    className="btn btn--play movie-card__button"
    href="/player/1"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </a>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="14"
      viewBox="0 0 18 14"
      width="18"
    >
      <use
        xlinkHref="#in-list"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
  <a
    className="btn movie-card__button"
    href="/films/1/review"
    onClick={[Function]}
  >
    Add review
  </a>
</div>
`;

exports[`Should render CardButtons component for main page correctly 1`] = `
<div
  className="movie-card__buttons"
>
  <a
    className="btn btn--play movie-card__button"
    href="/player/1"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </a>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="14"
      viewBox="0 0 18 14"
      width="18"
    >
      <use
        xlinkHref="#in-list"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
</div>
`;

exports[`Should render CardButtons component for non-favorite film correctly 1`] = `
<div
  className="movie-card__buttons"
>
  <a
    className="btn btn--play movie-card__button"
    href="/player/1"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </a>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="14"
      viewBox="0 0 18 14"
      width="18"
    >
      <use
        xlinkHref="#add"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
  <a
    className="btn movie-card__button"
    href="/films/1/review"
    onClick={[Function]}
  >
    Add review
  </a>
</div>
`;
